### DISCUSSION TITLE
Como aprender uma nova linguagem?

### DISCUSSION THREAD

**Discussion by viniciusmeneses** [frontendbr/forum#1144] [Discussão]
Pessoal, gostaria de saber como vocês acham que é o jeito ideal para aprender uma nova linguagem de programação por meio de cursos online/tutoriais? Deve-se escrever anotações para ajudar a relembrar? Primeiro anotar e depois escrever o código ou vice-versa?

Algumas pessoas dizem que para aprender a programar é necessário fazer anotações no caderno à mão, fazer anotações sobre o assunto no computador (talvez utilizando Markdown), comentar o código quase linha por linha explicando o que está acontecendo,  e ainda, algumas pessoas dizem que para aprender uma nova linguagem, não é necessário anotar, mas sim reescrever várias vezes o código, pois existe a documentação de linguagem como "anotação".

**Comment by giovannism20** [comment]
Na minha opinião, você mesmo já respondeu a sua pergunta. Tutoriais, ler sobre o assunto, ou seja, coisas que podem ser criadas com a linguagem e mesmo olhar projetos feitos com a linguagem no GitHub (ou em alguma outra plataforma), ajuda e evolui na sua forma não só de aprender a tecnologia, mas de evoluir suas técnicas de desenvolvimento e muitas vezes se manter atualizado.

Mas acho que cada um tem uma forma diferente de fazer isso tudo. Por exemplo, você comentou sobre a ideia de comentar em um código feito e comentar linha por linha, o que cada coisa faz. Talvez isso funcione para você e não para mim. Acho que o ideal mesmo é experimentar essas e outras para ver o que melhor se adequa a você. E mesmo essas imersão as vezes (que são esse conjunto de técnicas) como as citadas acima, pode ser suficiente para não escrever nem na mão, nem no seu próprio computador.

De qualquer forma, código comentado sempre é muito bom para outras pessoas entenderem. Para assim entender o que está sendo feito por ele ou mesmo você quando for fazer manutenção do mesmo. Que é diferente do que eu disse anteriormente.

Acho que deu para entender o que eu quis dizer. Qualquer dúvida, chama aí. ]]];),

**Comment by viniciusmeneses** [comment]
@GiovanniSM20 Obrigado pela ajuda! Concordo plenamente com você.

**Comment by felipefialho** [comment]
Linkando: #1062 #925 #769

**Comment by woliveiras** [comment]
Da uma olhada se este artigo te ajuda, @vini23hot [Como aprender uma nova linguagem de programação por ano](https://woliveiras.com.br/posts/Como-aprender-uma-nova-linguagem-de-programacao-por-ano/)

**Comment by AmandaFonseca** [comment]
Eu geralmente pego o material de estudo da linguagem que estou aprendendo e leio 1 vez para ir me familiarizando com o conteúdo, dou uma pausa e depois releio com muita atenção. Quando chega a hora de fazer um exercício  eu procuro fazer sem olhar a solução do professor se eu não conseguir fazer eu vejo  a solução do professor  e tento refazer o exercício outra hora ou outro dia até conseguir fazer sozinha ou sentir que assimilei a matéria bem.  Costuma dar certo para mim =)

**Comment by lussolli** [comment]
Eu prefiro ler a documentação das linguagens de programação que eu me interesso, e, então eu leio o código-fonte do exemplo e a explicação do autor.

Depois dessas leituras, eu copio o código-fonte no meu computador e executo ele. Se o programa executar sem nenhum problema, eu minimizo a aba com a documentação e apago o código-fonte do meu editor e tento escrever o programa novamente do zero sem olhar na documentação.
Se eu cometer um simples erro de digitação, eu apago o código-fonte novamente e tento fazer tudo do zero.

Depois que eu consigo memorizar bem a sintaxe da linguagem eu pego e fico alterando o programa original para descobrir como as coisas funcionam nessa linguagem.

Vez ou outra, quando eu estou no ônibus para ir ou voltar ao trabalho, eu pego um bloco de notas e fico escrevendo códigos na mão mesmo, para praticar tudo aquilo que eu tinha estudado.

As vezes eu acesso [essa](https://wiki.python.org.br/ListaDeExercicios) lista de exercícios nesse site sobre Python e tento resolver os problemas utilizando outras linguagens, como o PHP ou JavaScript.

**Comment by lussolli** [comment]
Acabei de me lembrar, o Atila do canal Nerdologia tinha postado um vídeo mostrando alguns "gatilhos" para as coisas mais facilmente, segue o link: https://www.youtube.com/watch?v=n5U83znAsbI
:)
